lib/logstash/outputs/tcp.rb

```diff
@@ -204,7 +176,38 @@ def server?
   end # def server?
 
   public
-  def receive(event)
-    @codec.encode(event)
-  end # def receive
+  def multi_receive_encoded(encoded)
+    if server?
+      @server_mutex.synchronize do
+        @client_threads.each do |client_thread|
+           encoded.each do |event,data|
+             client_thread[:client].write(data)
```

Comment:
I realize this was in the original, but maybe this would be a good time to rename `:client` to something more unique, like `:logstash_tcp_output_client`. These are global variables, so more namespacing = good.